File: engine/model/src/main/scala/tools/packet/movement/MovePet.scala
package tools.packet.movement

import net.opcodes.SendOpcode
import server.movement.LifeMovementFragment
import tools.packet.PacketInput

class MovePet(private var _characterId: Int, private var _petId: Int, private var _slot: Byte,
              private var _movementList: java.util.List[LifeMovementFragment]) extends PacketInput {
  def characterId: Int = _characterId

  def petId: Int = _petId

  def slot: Byte = _slot

  def movementList: java.util.List[LifeMovementFragment] = _movementList

  override def opcode(): SendOpcode = SendOpcode.MOVE_PET
}

File: engine/engine-model/src/main/java/tools/packet/movement/MovePet.java
package tools.packet.movement;

import java.util.List;

import net.opcodes.SendOpcode;
import server.movement.LifeMovementFragment;
import tools.packet.PacketInput;

public record MovePet(int characterId, int petId, byte slot,
                      List<LifeMovementFragment> movementList) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MOVE_PET;
   }
}